src/frontend/core/executor/impl/TriangleCountExecutor.cpp

```diff
@@ -34,13 +50,400 @@ TriangleCountExecutor::TriangleCountExecutor(SQLiteDBInterface *db, PerformanceS
     this->request = jobRequest;
 }
 
+static void allocate(int p, string w, std::map<int, string> &alloc, std::set<int> &remain,
+                     std::map<int, std::vector<string>> &p_avail, std::map<string, int> &loads) {
+    alloc[p] = w;
+    remain.erase(p);
+    p_avail.erase(p);
+    loads[w]++;
+    if (loads[w] >= 3) {
+        for (auto it = p_avail.begin(); it != p_avail.end(); it++) {
+            auto &ws = it->second;
+            auto itr = std::find(ws.begin(), ws.end(), w);
+            if (itr != ws.end()) {
+                ws.erase(itr);
+            }
+        }
+    }
+}
+
+static int get_min_partition(std::set<int> &remain, std::map<int, std::vector<string>> &p_avail) {
+    int p0 = *remain.begin();
+    size_t m = 1000000000;
```

Comment:
Can we rename the variables `p, m, ws` into meaningful ones?

Reply:
Done

Reply:
Ok